test/hamiltonian/test_heat.py

```diff
@@ -0,0 +1,304 @@
+
+        ref_circ = circuit.QuantumCircuit(2, 1)
+        ref_circ.x(0)
+        ref_circ.ry(np.pi / 2, 1)
+        ref_circ.barrier()
+        ref_circ.append(exp.experiment_options.heat_gate, [0, 1])
+        ref_circ.z(1)
+        ref_circ.barrier()
+        ref_circ.append(exp.experiment_options.heat_gate, [0, 1])
+        ref_circ.z(1)
+        ref_circ.barrier()
+        ref_circ.rx(np.pi / 2, 1)
+        ref_circ.measure(1, 0)
+
+        self.assertEqual(heat_circ, ref_circ)
+
+
+@ddt
+class TestZXHeat(QiskitExperimentsTestCase, HeatExperimentsTestCase):
+    """Test ZX Heat experiment."""
+
+    @staticmethod
+    def create_generator(
+        angle=np.pi / 2,
+        e_zx=0.0,
+        e_zy=0.0,
+        e_zz=0.0,
+        e_ix=0.0,
+        e_iy=0.0,
+        e_iz=0.0,
+    ):
+        """Create generator Hamiltonian represented by numpy array."""
+        generator_ham = (
+            0.5
+            * (
+                (angle + e_zx) * qi.Operator.from_label("XZ")
+                + e_zy * qi.Operator.from_label("YZ")
+                + e_zz * qi.Operator.from_label("ZZ")
+                + e_ix * qi.Operator.from_label("XI")
+                + e_iy * qi.Operator.from_label("YI")
+                + e_iz * qi.Operator.from_label("ZI")
+            ).data
+        )
+
+        return generator_ham
+
+    @data(
+        [0.08, -0.01],
+        [-0.05, 0.13],
+        [0.15, 0.02],
+        [-0.04, -0.02],
+        [0.0, 0.12],
+        [0.12, 0.0],
+    )
+    @unpack
+    def test_x_error_amplification(self, e_zx, e_ix):
+        """Test for X error amplification."""
+        exp = ZX90HeatXError(qubits=(0, 1), backend=self.backend)
+        generator = self.create_generator(e_zx=e_zx, e_ix=e_ix)
+        gate = self.create_heat_gate(generator)
+        exp.set_experiment_options(heat_gate=gate)
+        exp.set_transpile_options(basis_gates=["x", "sx", "rz", "unitary"])
+
+        exp_data = exp.run().block_for_results()
+
+        self.assertAlmostEqual(exp_data.analysis_results("A_IX").value.value, e_ix, delta=0.01)
+        self.assertAlmostEqual(exp_data.analysis_results("A_ZX").value.value, e_zx, delta=0.01)
+
+    @data(
+        [0.02, -0.01],
+        [-0.05, 0.03],
+        [0.03, 0.02],
+        [-0.04, -0.01],
+        [0.0, 0.01],
+        [0.01, 0.0],
+    )
+    @unpack
+    def test_y_error_amplification(self, e_zy, e_iy):
+        """Test for Y error amplification."""
+        exp = ZX90HeatYError(qubits=(0, 1), backend=self.backend)
+        generator = self.create_generator(e_zy=e_zy, e_iy=e_iy)
+        gate = self.create_heat_gate(generator)
+        exp.set_experiment_options(heat_gate=gate)
+        exp.set_transpile_options(basis_gates=["x", "sx", "rz", "unitary"])
+
+        exp_data = exp.run().block_for_results()
+
+        # larger error torelance to allow commutator term
+        self.assertAlmostEqual(exp_data.analysis_results("A_IY").value.value, e_iy, delta=0.05)
```

Comment:
This delta is pretty big -- as big as the biggest input error. In practice, we saw that the fitted error was around 0.7 of the actual error. Could we tighten the delta a bit? Maybe `0.5 * e_iy` (or compare to `0.7 e_iy` and make the delta even smaller?)?

Reply:
This is good suggestion. Done in https://github.com/Qiskit/qiskit-experiments/pull/625/commits/e379b49062b923f159e197b5715aacf8494956d3

Reply:
Hmm, that is better, but couldn't you do something like:

```python
        self.assertAlmostEqual(
            exp_data.analysis_results("A_IZ").value.value,
            e_iz * 0.7,
            delta=max(0.1 * abs(e_iz), 0.01),
        )
```

I am not sure if the 0.7 factor is right but you found that the inferred errors were quite linear actual error. If you use this, it would be good to put a comment in saying that the factor was found by numerical calculation and deviates from 1.0 due to anticommutation between the error terms and the ZX rotation in case someone looks at this later.

Reply:
It worked and sees like more precise. I hope this works in every platform.
https://github.com/Qiskit/qiskit-experiments/pull/625/commits/9819c16a15e22bd49f1365ae3a663cc9a38d1bef